src/Keystore/Account.cpp

```diff
@@ -24,6 +25,10 @@ namespace CodingKeys {
 } // namespace CodingKeys
 
 Account::Account(const nlohmann::json& json) {
+    if (json.find(CodingKeys::derivation) != json.end()) {
+        derivation = TWDerivation(json[CodingKeys::derivation].get<uint32_t>());
+    }
+
     if (json[CodingKeys::derivationPath].is_object()) {
```

Comment:
```suggestion
    } else if (json[CodingKeys::derivationPath].is_object()) {
```

Reply:
Previous if is for derivation field, this is for derivationPath, should be independent.